common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -947,23 +721,23 @@ public AsciiString trim() {
     /**
      * Compares a {@code CharSequence} to this {@code String} to determine if their contents are equal.
      *
-     * @param cs the character sequence to compare to.
+     * @param a the character sequence to compare to.
      * @return {@code true} if equal, otherwise {@code false}
      */
-    public boolean contentEquals(CharSequence cs) {
-        if (cs == null) {
-            throw new NullPointerException();
+    public boolean contentEquals(CharSequence a) {
+        if (this == a || a.length() != length()) {
```

Comment:
I'm a bit confused by this... return true here if `a.length() != length()` looks not right to me.